doc/i2c_bus_lock_prevention/i2c_bus_lock_prevention_hld.md

```diff
@@ -0,0 +1,425 @@
+
+#### 7.1. The Method to Recover I2C Bus Lock and the Limitation of the Design
+
+The design relies on the i2c mux devices to isolate the faulty device on the bus. If the faulty device is not behind any i2c mux device, there is no way to isolate the device and the i2c bus cannot be recovered by this design.
+
+It is assumes that only swappable devices could cause i2c bus lock. So this feature will not handle the case that the faulty device is a fixed i2c device.
+
+Take the i2c tree example below to explain the method to detect I2C bus lock and recover it.
+The scenarios that the recover cannot be done are also illustrated.
+
+![i2c tree example 2, detect and recover](i2c_tree_example2.png)
+###### Figure 2: I2C Tree Example 2
+The i2c devices with color are located behind a channel of a i2c mux device.
+The same color indicates the same i2c isolated region.
+
+Scenario 1: I2C Dev2 is in fault state (Recovery is possible)
+    Given that I2C Dev 2 is the faulty device.
+    Here are the brief illustrated steps for the key events and actions.
+    1. I2C Dev 2 pulls the i2c signal line permanently and results in i2c bus lock.
+    2. CPU detects the i2c bus lock, it takes the action to reset I2C Mux1 and I2C Mux2.
+    3. All channels are closed on I2C Mux1 and I2C Mux2. The faulty device I2C Dev2 will not impact the signal lines before I2C Mux1. So CPU is able to operate on the bus now.
+    4. I2C Dev 2 needs to put into the isolation list.
+
+Scenario 2: I2C Dev1 is in fault state (Recovery is impossible)
+Given that I2C Dev 1 pulls the I2C signal line.
+Here are the brief illustrated steps for the key events and actions.
+    1. I2C Dev 1 pulls the I2C signal line permanently.
+    2. CPU detects the I2C bus lock, it takes the action to reset I2C Mux1 and I2C Mux2.
+    3. I2C Dev 1 is not located behind I2C Mux1 and I2C Mux2 so the faulty state cannot be isolated/recovered.
+
+Scenario 3: I2C Dev 6 is in fault state (Recovery is impossible)
+Given that I2C Dev 6 pulls the I2C signal line.
+Here are the brief illustrated steps for the key events and actions.
+    1. I2C Dev 6 pulls the I2C signal line permanently.
+    2. CPU detects the I2C bus lock, it takes the action to reset I2C Mux1 and I2C Mux2.
+    3. All channels are closed on I2C Mux1 and I2C Mux2. The faulty device I2C Dev6 will not impact the signal lines before I2C Mux1. So CPU is able to operate on the bus now.
+    4. I2C Dev 6 and 7 need to put into the isolation list.
+
+#### 7.2. The Method to Detect I2C Bus Lock
+
+On a specific hardware model, choose three representative i2c devices to poll.
+
+The criteria to choose the representative i2c devices are shown below.
+* In order to minimize the impact to the system performance, the chosen i2c device is seldom visited by the platform driver daemons.
+* The polling interval shall be defined to a value that will not impact the system performance severely.
+

```

Comment:
Consequently, if a representative device must be a transceiver, the `i2chealthd` daemon will compete for access with the `xcvrd` daemon. 

The same could apply to fans and `thermalctld`